base/compiler/optimize.jl

```diff
@@ -374,16 +391,38 @@ function convert_to_ircode(ci::CodeInfo, code::Vector{Any}, coverage::Bool, sv::
                 inbounds_depth -= 1
             end
             stmt = nothing
+        elseif isexpr(stmt, :inline)
+            if stmt.args[1]::Bool
+                push!(inline_flags, true)
+            else
+                pop!(inline_flags)
+            end
+            stmt = nothing
+        elseif isexpr(stmt, :noinline)
+            if stmt.args[1]::Bool
+                push!(inline_flags, false)
+            else
+                pop!(inline_flags)
+            end
+            stmt = nothing
         else
             stmt = normalize(stmt, meta)
         end
         code[i] = stmt
-        if !(stmt === nothing)
+        if stmt !== nothing
             if inbounds_depth > 0
                 flags[i] |= IR_FLAG_INBOUNDS
             end
+            if !isempty(inline_flags)
+                if last(inline_flags)
+                    flags[i] |= IR_FLAG_INLINE
+                else
+                    flags[i] |= IR_FLAG_NOINLINE
+                end
+            end
         end
     end
```

Comment:
I wonder if it would be better process all of these flags better during `jl_code_info_set_ir`, since we already pre-process a number of other flags there, as part of converting from unattached lowered form to associated source code form.

Reply:
Can we just pack it into `CodeInfo.ssaflags` ?

Reply:
Yes. We currently don't.

Reply:
what do you think on ff0b8a9cdb90568d34a1a42180bb6a5877e69b30 ?